### DISCUSSION TITLE
Keep Capital Letters in Response Header?

### DISCUSSION THREAD

**Discussion by Expugn** [vercel/community#2805] [Help] [answered]
### Summary

Hi,

I'm using [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions) and would like to send a response header that keeps its capital letters (e.g. `RESPONSE-HEADER`). Unfortunately, Vercel seems to turn `RESPONSE-HEADER` into `Response-Header` which causes issues for my use-case.

I've tried the following options, all of which return the undesired `Response-Header` instead of `RESPONSE-HEADER`:
API Routes with Next.js
```js
export default (req, res) => {
    res.setHeader("RESPONSE-HEADER", "value");
    res.status(200).json({ code: 200 });
};
```

Setting header in `vercel.json`
```json
{
  "headers": [
    {
      "source": "/api/example",
      "headers": [
        {
          "key": "RESPONSE-HEADER",
          "value": "value"
        }
      ]
    }
  ]
}
```

Standalone Express
```js
const app = require('express')();

app.get('/api', (req, res) => {
    res.setHeader("RESPONSE-HEADER", "value");
    res.status(200).json({ code: 200 });
});
```

Is there a way I can configure Vercel to keep my preferred response header name without modifying it?

Thanks in advance.

### Steps to Reproduce

1. Start a new Vercel project and set up Serverless Functions or Express.
2. Set response header to something that's all capital letters, e.g. `RESPONSE-HEADER`.
3. Deploy and check response. Expected: `RESPONSE-HEADER` ; Given: `Response-Header`

**Comment by smaeda-ks** [comment, marked as answer]
@Expugn while I understand your point, you should not implement such a feature where it expects HTTP header names to be case-sensitive. RFC states that header names are case-insensitive:
https://www.rfc-editor.org/rfc/rfc9110#section-5

We do not plan to change this behavior for that reason.

**Reply by Expugn** [reply]
Fair enough, thanks for the response!
